packages/aws-cdk-lib/aws-codebuild/lib/fleet.ts

```diff
@@ -0,0 +1,197 @@
+import { Construct } from 'constructs';
+import { CfnFleet } from './codebuild.generated';
+import { ComputeType } from './compute-type';
+import { EnvironmentType } from './environment-type';
+import { Arn, ArnFormat, IResource, Resource, Token } from '../../core';
+
+/**
+ * Construction properties of a CodeBuild {@link Fleet}.
+ */
+export interface FleetProps {
+  /**
+   * The name of the Fleet.
+   *
+   * @default - CloudFormation generated name
+   */
+  readonly fleetName?: string;
+
+  /**
+   * The number of machines allocated to the compute ﬂeet.
+   * Deﬁnes the number of builds that can run in parallel.
+   *
+   * Minimum value of 1.
+   */
+  readonly baseCapacity: number;
+
+  /**
+   * The instance type of the compute fleet.
+   */
+  readonly computeType: FleetComputeType;
+
+  /**
+   * The environment type of the fleet.
+   */
+  readonly environmentType: EnvironmentType;
+}
+
+/**
+ * Represents a {@link Fleet} for a reserved capacity CodeBuild project.
+ */
+export interface IFleet extends IResource {
+  /**
+   * The ARN of the fleet.
+   * @attribute
+   */
+  readonly fleetArn: string;
+
+  /**
+   * The name of the fleet.
+   * @attribute
+   */
+  readonly fleetName: string;
+
+  /**
+   * The compute type of the fleet.
+   */
+  readonly computeType: FleetComputeType;
+
+  /**
+   * The environment type of the fleet.
+   */
```

Comment:
```suggestion
  /**
   * The operating system and/or chip type available to projects spawned in this fleet.
   */
```

Reply:
I wouldn't say that the projects are spawned within the fleet, more like the build projects are using the dedicated fleet instances instead of on-demand resources.
How about: `The build environment (operating system/architecture/accelerator) type made available to projects using this fleet`?

Reply:
:+1: that's very clear, please add that

Reply:
Done